### PR TITLE
remove deprecated slicing semantics of polynomials

### PR DESCRIPTION
**Pull request by fchapoton**
### 📚 Description

This removes the deprecations introduced in #18940. So the only allowed slicing of polynomials is `p[:k]`.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/laurent_series_ring_element.pyx [hunk 1]
```diff
@@ -523,22 +523,32 @@ cdef class LaurentSeries(AlgebraElement):
...
-            See https://github.com/sagemath/sage/issues/18940 for details.
+            sage: f[:2]
             -5*t^-10 + 1/3 + t + O(t^5)
+
+        Any other kind of slicing is an error, see :trac:`18940`::
+
+            sage: f[-10:2:2]
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: polynomial slicing with a step is not defined
```
Comment (mkoeppe) [inline src/sage/rings/laurent_series_ring_element.pyx]: Shouldn't this be `IndexError` or something like this?

Comment (fchapoton) [reply src/sage/rings/laurent_series_ring_element.pyx]: not sure if this would be any better

Comment (mkoeppe) [reply src/sage/rings/laurent_series_ring_element.pyx]: I read `NotImplementedError` as an invitation to implement this feature.

But here a feature has been deliberately removed.

**Comment by mkoeppe** [comment]
One doctest output needs updating - https://github.com/sagemath/sage/actions/runs/4347013220/jobs/7593799402#step:11:7798

**Review by mkoeppe — APPROVED** [review]

**Pull request merged**